ArduCopter/ReleaseNotes.txt

```diff
@@ -1,5 +1,432 @@
+- Temperature sensor bindings added
+- uint64 support
+- VTOL-quicktune.lua minor enhancements including increasing YAW_FLTE_MAX to 8
+- x-quad-cg-allocation.lua applet corrects for the CoM discrepancy in a quad-X frame
+
+13) GCS / mavlink related changes and fixes
+
+- BATTERY2 message deprecated (replaced by BATTERY_STATUS)
+- CMD_MISSION_START/STOP rejected if first-item/last-item args provided
+- Deny attempts to upload two missions simultaneously
+- ESC telemetry works even when less than 4 ESCs
+- Fence and Rally points may be uploaded using FTP
+- GPS_INPUT and HIL_GPS handles multiple GPSs
+- HIGHRES_IMU mavlink message support (used by companion computers to receive IMU data from FC)
+- MAV_CMD_COMPONENT_ARM_DISARM accepts force arm magic value of 21196
+- MAV_CMD_DO_SET_SAFETY_SWITCH_STATE support
+- MAV_CMD_SET_HAGL support (Plane only)
+- MAVFTP respects TX buffer flow control to improve FTP on low bandwidth links
+- MAVLink receiver support (RADIO_RC_CHANNELS mavlink message)
+- Message interval supports TERRAIN_REPORT msg
+- Mission upload may be cancelled using MISSION_CLEAR_ALL
+- MOUNT_CONFIGURE, MOUNT_CONTROL messages deprecated
+- RC_CHANNELS_RAW deprecated
+- Serial passthrough supports parity allowing STM32CubeProgrammer to be used to program FrSky R9 receivers
+- SET_ATTITUDE_TARGET angular rate input frame fixed (Copter only)
+- TIMESYNC and NAMED_VALUE_FLOAT messages not sent on high latency MAVLink ports
+
+14) Logging enhancements and fixes
+
+- AC_PID resets and I-term sets logged
+- ANG provides attitude at high rate (equivalent to ATT)
+- ATT logs angles as floats (better resolution than ints)
+- CAND message gets driver index
+- DCM log message includes roll/pitch and yaw error
+- EDT2 log msg includes stress and status received via extended DShot Telemetry v2
+- EFI ECYL cylinder head and exhaust temp logs in degC (was Kelvin)
+- ESCX log msg includes DroneCAN ESCs status, temp, duty cycle and power pct
+- FMT messages written as required instead of all at beginning
+- Logging restarted after download completes when LOG_DISARMED = 1
+- MISE msg logs active mission command (CMD logged when commands are uploaded)
+- ORGN message logging fixed when set using SET_GPS_GLOBAL_ORIGIN
+- RPM sensor logging gets instance field, quality and health fields
+- Short filename support removed (e.g log1.BIN instead of 00000001.BIN)
+- Temperature sensor logging option for only sensors with no source (see TEMP_LOG)
+- UART data rates logged at 1hz (see UART message)
+
+15) ROS2 / DDS support
+
+- Airspeed published
+- Battery topic reports all available batteries
+- Compile-time configurable rates for each publisher
+- DDS_TIMEOUT_MS and DDS_MAX_RETRY set timeout and num retries when client pings XRCE agent
+- GPS global origin published at 1 Hz
+- High frequency raw imu data transmission
+- Joystick support
+- Support sending waypoints to Copter and Rover
```

Comment:
```suggestion
- Support sending waypoints to Copter and Rover
- Remove the XML refs file in favor of binary entity creation
```

From 296cdc58a3b43e8cbe199d6feef42bc583073702. This is an important breaking change.